colophon.py

```diff
@@ -53,22 +53,31 @@ def latest_dataset():
         return
 
     path = URL_BASE + match.group(1)
-    last_modified = datetime.strptime(match.group(2), '%Y-%m-%dT%H:%M:%S.%f%z')
+    last_modified = datetime.strptime(match.group(2), '%Y-%m-%dT%H:%M:%S.%fZ')
     size_in_gb = int(match.group(3)) / 1_000_000_000
 
     return path, last_modified, size_in_gb
 
 def prompt_download(local_data_path):
-    print('No Unpaywall dataset found. Searching online...')
+    print('No local Unpaywall dataset found. Searching online...')
     path, last_modified, size_in_gb = latest_dataset()
     if path:
         print(f'Dataset found. Last update: {last_modified:%d %b %Y}.')
         if input(f'Download this {size_in_gb:1.1f} GB dataset now? [Y/n] ').lower() != 'n':
             response = requests.get(path, stream=True)
-            with open(local_data_path, 'wb') as handle:
-                for data in tqdm(response.iter_content()):
-                    handle.write(data)
-            print('Done! Proceeding...')
+            if response.ok:
+                with open(local_data_path, 'wb') as handle:
+                    with tqdm(
+                        unit='B', unit_scale=True, unit_divisor=1000,
+                        total=int(response.headers.get('content-length', 0))
+                    ) as pbar:
+                        for chunk in response.iter_content(chunk_size=4096):
+                            handle.write(chunk)
+                            pbar.update(len(chunk))
+                print('Done! Proceeding...')
+            else:
+                print('Non-OK response: status code {response.status_code}')

```

Comment:
Should this be an f-string, like `f'Non-OK response: status code {response.status_code}'`?

Reply:
Oops, yes! Thanks for catching it :)